uaclient/tests/test_cli_status.py

```diff
@@ -818,6 +853,122 @@ def test_unicode_dash_replacement_when_unprintable(
         expected_out = ATTACHED_STATUS.format(dash=expected_dash, notices="")
         assert expected_out == out
 
+    @pytest.mark.parametrize(
+        "exception_to_throw,exception_type,exception_message",
+        (
+            (
+                exceptions.UrlError("Not found", 404),
+                exceptions.UrlError,
+                "Not found",
+            ),
+            (
+                exceptions.ContractAPIError(
+                    exceptions.UrlError("Unauthorized", 401),
+                    {"message": "unauthorized"},
+                ),
+                exceptions.UserFacingError,
+                "Invalid token. See https://ubuntu.com/advantage",
+            ),
+        ),
+    )
+    def test_errors_are_raised_appropriately(
+        self,
+        _m_getuid,
+        m_get_contract_information,
+        _m_get_avail_resources,
+        _m_should_reboot,
+        _m_remove_notice,
+        exception_to_throw,
+        exception_type,
+        exception_message,
+        capsys,
+        FakeConfig,
+    ):
+        """Check that simulated status json/yaml output raises errors."""
+
+        m_get_contract_information.side_effect = exception_to_throw
+
+        cfg = FakeConfig()
+
+        args = mock.MagicMock(
+            format="json", all=False, simulate_with_token="some_token"
+        )
+
+        with pytest.raises(Exception) as exc:
```

Comment:
Should we use `exception_type` here for the `raises` parameter ?

Reply:
done